packages/app-bridge/src/react/useBlockTemplates.ts

```diff
@@ -21,21 +22,42 @@ export const useBlockTemplates = (appBridge: AppBridgeBlock) => {
         }
     };
 
+    const handleErrorMessage = (error: unknown) => {
+        let errorMessage;
+
+        if (error instanceof Error) {
+            errorMessage = error.message;
+        } else if (typeof error === 'string') {
+            errorMessage = error;
+        } else {
+            errorMessage = String(error);
+        }
+
+        setErrorMessage(errorMessage);
+    };
+
     // Fetch the block templates on mount.
     // And add listener for block template updates.
     useEffect(() => {
         let componentMounted = true;
 
         if (blockId) {
             const mountingFetch = async () => {
-                const allBlockTemplates = await appBridge.getBlockTemplates();
-                if (componentMounted) {
-                    setBlockTemplates(allBlockTemplates);
-                }
+                appBridge
+                    .getBlockTemplates()
+                    .then((result) => {
+                        if (componentMounted) {
+                            setBlockTemplates(result);
+                        }
+                    })
+                    .catch((error) => {
+                        handleErrorMessage(error);
+                    });
             };
 
-            mountingFetch();
-            window.emitter.on('AppBridge:BlockTemplatesUpdated', updateBlockTemplatesFromEvent);
+            mountingFetch().then(() =>
```

Comment:
`async`/`await` please 😬

Reply:
Sorry, I'm not getting it. What's wrong with `.then()`?

Reply:
Our coding style is to use `async`/`await`

Reply:
Ah, got it. Sorry about not following the coding style 🙈